usr/share/rear/prep/GNU/Linux/210_include_dhclient.sh

```diff
@@ -31,11 +32,40 @@ dhcp_interfaces_active() {
         # include DHCP CLIENT support in Relax-and-Recover
         Log "Running DHCP client found, enabling USE_DHCLIENT"
         USE_DHCLIENT=y
+        return  # no additional checks needed
+    fi
+
+    # Strategy 2: Check if systemd-networkd is active and configured to use its built-in DHCP client
+    if type systemctl &>/dev/null && systemctl is-active --quiet systemd-networkd; then
+        # Check the systemd.network(5) configuration for an effective configuration file enabling DHCP
+        # for some network interface. Configuration files live in {/etc,/run,/lib}/systemd/network/*.network and
+        # systemd configuration file priorities apply. Note that we do not check for additional *.conf files in
+        # "drop-in" directories (*.network.d) as this would add an additional layer of complexity.
+        # (Life could be easier if there was something like 'systemctl show *.network', but there isn't yet.)
+        local config_bases=(/etc /run /lib)  # priority: high to low
+        local config_files=()  # relative paths to config bases
+        local config_base config_file
+
+        # Read relative config file paths into the 'config_files' array variable (safe in case of blanks).
+        while IFS='' read -r config_file; do config_files+=("$config_file"); done < <(for config_base in "${config_bases[@]}"; do (cd "$config_base" && find systemd/network -name '*.network' -print) done | sort -u)
+
+        for config_file in "${config_files[@]}"; do
+            for config_base in "${config_bases[@]}"; do
+                if [[ -f "$config_base/$config_file" ]]; then
+                    if grep -Eq '^[[:space:]]*DHCP[[:space:]]*=[[:space:]]*(yes|ip)' "$config_base/$config_file"; then
+                        Log "Detected an active systemd-networkd configured to use its built-in DHCP client, enabling USE_DHCLIENT"
+                        USE_DHCLIENT=y
+                        return  # no additional checks needed
+                    fi
+                    break  # do not consider lower-priority config files
```

Comment:
@OliverO2 
I am uncertain why lower-priority config files are not considered.
I guess this is because it is about how systemd evaluates config files
where higher-priority config files overwrite all what is in lower-priority config files
according to what you wrote in
https://github.com/rear/rear/issues/2375#issue-601938141
```
In case of systemd-networkd, if netplan had generated
a configuration file /run/systemd/network/10-netplan-eno1.network,
it could be overridden by creating a configuration file
/etc/systemd/network/10-netplan-eno1.network.
```
which means "completely overridden" and
not something like "combined" where entries
in higher-priority config files overwrite already
existsing entries in lower-priority config files.

Reply:
Exactly, higher-priority files completely replace lower-priority ones. [systemd\.network(5)](http://manpages.ubuntu.com/manpages/bionic/en/man5/systemd.network.5.html):

>  The .network files are read from the files located in the system network directory
       /lib/systemd/network, the volatile runtime network directory /run/systemd/network and the
       local administration network directory /etc/systemd/network. All configuration files are
       collectively sorted and processed in lexical order, regardless of the directories in which
       they live. However, files with identical filenames replace each other. Files in /etc have
       the highest priority, files in /run take precedence over files with the same name in /lib.
       This can be used to override a system-supplied configuration file with a local file if
       needed. As a special case, an empty file (file size 0) or symlink with the same name
       pointing to /dev/null disables the configuration file entirely (it is "masked").